indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/common/RecordSupplier.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.seekablestream.common;
+
+import com.google.common.annotations.Beta;
+
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * The RecordSupplier interface is a wrapper for the incoming seekable data stream
+ * (i.e. Kafka consumer, Kinesis streams)
+ *
+ * @param <T1> Partition Number Type
+ * @param <T2> Sequence Number Type
+ */
+@Beta
+public interface RecordSupplier<T1, T2> extends Closeable
+{
+  /**
+   * assigns a set of partitions to this RecordSupplier
```

Comment:
repeating the method signature word for word. Such comments are not really needed, in fact they introduce more issues, imaging tomorrow someone refactor the code and change the name of method or name of the class `RecordSupplier` this comment will be inaccurate.